app/controllers/api/v1/payments_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# API for payments
+class Api::V1::PaymentsController < Api::V1::ApiController
+  before_action :set_payment, only: %i[show update destroy]
+
+  # GET /payments
+  def index
+    @payments = Payment.all
+
+    render json: @payments
+  end
+
+  # GET /payments/:slug
+  def show
+    render json: @payment
+  end
+
+  # POST /payments
+  def create
+    @payment = Payment.new(payment_params)
+
+    if @payment.save
+      render json: @payment, status: :created, location: @payment
+    else
+      render json: @payment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /payments/:slug
+  def update
+    if @payment.update(payment_params)
+      render json: @payment
+    else
+      render json: @payment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /payments/:slug
+  def destroy
+    @payment.destroy
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_payment
+    @payment = Payment.find_by!(slug: params[:slug])
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # Only allow a trusted parameter "white list" through.
+  def payment_params
+    params.require(:payment).permit(:paid_on,
+                                    :care_started_on,
+                                    :care_finished_on,
+                                    :amount,
```

Comment:
Not to be nitpicky, but if we have to send `..._cents`, will `amount` or `discrepancy` ever be sent as params?

Reply:
I thought about this, too.  (This isn't being nitpicky. At least not in my book.)
It is legitimate to interact with the attribute without the `_cents` or `_currency` suffixes, e.g. in specs or non-strict API, as in a rake task.

However, we can certainly restrict interaction to only the `_cents` and `_currency` -suffixed parameters for now.
It's your call.  Either way is fine.  And I can make a note in a block comment for this method so it's clear either way.

Reply:
Ahhh, you know what, you make a good point, I wasn't thinking about raw access to the controller not via the API.  I'm good with leaving them both in; maybe add in the _currency too while we're add it?  Never know if we'll want to change it........ 🤔